numpy/core/numeric.py

```diff
@@ -1896,24 +1896,24 @@ def fromfunction(function, shape, **kwargs):
 
     See Also
     --------
-    indices, meshgrid
+    indices, meshgrid, vectorize
 
     Notes
     -----
     Keywords other than `dtype` are passed to `function`.
 
     Examples
     --------
-    >>> np.fromfunction(lambda i, j: i == j, (3, 3), dtype=int)
-    array([[ True, False, False],
-           [False,  True, False],
-           [False, False,  True]])
-
     >>> np.fromfunction(lambda i, j: i + j, (3, 3), dtype=int)
     array([[0, 1, 2],
            [1, 2, 3],
            [2, 3, 4]])
 
+    >>> np.fromfunction(np.vectorize(lambda i, j: i == j), (3, 3), dtype=int)
```

Comment:
Why the `np.vectorize` here?

Reply:
Using a vectorized function in fromfunction produces the same output as would be expected from the original documentation; ie. can use a function that given a pair of coordinates returns the desired value at that position of the array. I think this is likely the output for which people who found fromfunction were looking.
I thought it would be a useful note, but I can remove it if it seems extraneous.

Reply:
This example doesn't really show that though, because `np.fromfunction(lambda i, j: i == j, (3, 3), dtype=int)` gives the same result.

Reply:
Perhaps you should have an example where you print both arguments